aptos-move/block-executor/src/scheduler.rs

```diff
@@ -570,6 +571,38 @@ impl Scheduler {
         SchedulerTask::NoTask
     }
 
+    pub fn finish_execution_during_commit(&self, txn_idx: TxnIndex) {
+        // we have exclusivity on this transaction
+
+        let txn_deps: Vec<TxnIndex> = {
```

Comment:
Let's factor out this, resumption & execution idx handling to a helper method and re-use with finish_execution?